include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1736,15 +1736,22 @@ __pattern_rotate_copy(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Bid
 }
 
 template <typename _BackendTag, typename _ExecutionPolicy, typename _ForwardIterator1, typename _ForwardIterator2,
-          typename _OutputIterator, typename _Compare, typename _IsOpDifference>
+          typename _OutputIterator, typename _Compare, typename _SetTag>
 _OutputIterator
 __pattern_hetero_set_op(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _ForwardIterator1 __first1,
                         _ForwardIterator1 __last1, _ForwardIterator2 __first2, _ForwardIterator2 __last2,
-                        _OutputIterator __result, _Compare __comp, _IsOpDifference __is_op_difference)
+                        _OutputIterator __result, _Compare __comp, _SetTag __set_tag)
 {
     typedef typename std::iterator_traits<_ForwardIterator1>::difference_type _Size1;
 
     const _Size1 __n1 = __last1 - __first1;
+    _Size1 __output_size = __n1;
+    if constexpr (_SetTag::__is_one_shot_v)
+    {
+        const _Size1 __n2 = __last2 - __first2;
```

Comment:
Since `__last2` and `__first2` are both forward iterators, we might want to use `std::distance` here.